core/store/src/trie/shard_tries.rs

```diff
@@ -442,22 +471,69 @@ impl KeyForStateChanges {
         key
     }
 
+    /// Without changing the existing TrieKey format, encodes ShardUId into the row key.
+    /// See `delayed_receipt_key_from_trie_key` for decoding this row key.
+    pub fn delayed_receipt_key_from_trie_key(
+        block_hash: &CryptoHash,
+        trie_key: &TrieKey,
+        shard_uid: &ShardUId,
+    ) -> Self {
+        let mut key = Self::new(block_hash, trie_key.len());
+        trie_key.append_into(&mut key.0);
+        key.0.extend(shard_uid.version.to_le_bytes());
+        key.0.extend(shard_uid.shard_id.to_le_bytes());
+        key
+    }
+
+    /// Extracts ShardUId from row key which contains ShardUId encoded.
+    /// See `delayed_receipt_key_from_trie_key` for encoding ShardUId into the row key.
+    pub fn delayed_receipt_key_decode_shard_uid(
+        row_key: &[u8],
+        block_hash: &CryptoHash,
+        trie_key: &TrieKey,
+    ) -> Result<ShardUId, KeyForStateChangesError> {
+        let prefix = KeyForStateChanges::from_trie_key(block_hash, trie_key);
+        let prefix = prefix.as_ref();
+
+        let suffix = &row_key[prefix.len()..];
+        if suffix.is_empty() {
+            return Err(KeyForStateChangesError::DelayedReceiptRowKeyError(
+                row_key.to_vec(),
+                trie_key.clone(),
+            ));
+        }
+        let version_chunk = &suffix[..std::mem::size_of::<ShardVersion>()];
```

Comment:
this code (505-523) should be put in a method of ShardUId ('deserialize_from_key'?)

And similarly - line 483 + 484 should be moved to ShardUid method too.

Reply:
Removed, as that code is already available as `to_bytes()` and `try_from`